crates/sparrow-batch/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+#![warn(
+    rust_2018_idioms,
+    nonstandard_style,
+    future_incompatible,
+    clippy::mod_module_files,
+    clippy::print_stdout,
+    clippy::print_stderr
+)]
+
+mod batch;
+mod error;
+mod row_time;
+
+pub use batch::*;
+pub use error::Error;
+pub use row_time::*;
+
+#[cfg(any(test, feature = "testing"))]
```

Comment:
Just to call it out since we discussed. This is a pattern that adds the `testing` module if either:

1. We're in this crate and compiling for tests `#[cfg(test)]`
2. OR, we're outside this crate and have the `testing` feature enabled.

We could then have *other* crates turn on the `testing` dependency if they have the `test` cfg. But either way means that we can have these used in other places, but somewhat clear that it's only for tests.